src/transformers/models/textnet/modeling_textnet.py

```diff
@@ -0,0 +1,552 @@
+        else:
+            self.vertical_conv, self.vertical_batch_norm = None, None
+
+        if kernel_size[0] != 1:
+            self.horizontal_conv = nn.Conv2d(
+                in_channels=in_channels,
+                out_channels=out_channels,
+                kernel_size=(1, kernel_size[1]),
+                stride=stride,
+                padding=hor_pad,
+                bias=False,
+            )
+            self.horizontal_batch_norm = nn.BatchNorm2d(num_features=out_channels)
+        else:
+            self.horizontal_conv, self.horizontal_batch_norm = None, None
+
+        self.rbr_identity = (
+            nn.BatchNorm2d(num_features=in_channels) if out_channels == in_channels and stride == 1 else None
+        )
+
+    def forward(self, hidden_states):
+        if self.training:
+            if hasattr(self, "fused_conv"):
+                self.__delattr__("fused_conv")
+
+            main_outputs = self.main_conv(hidden_states)
+            main_outputs = self.main_batch_norm(main_outputs)
+            if self.vertical_conv is not None:
+                vertical_outputs = self.vertical_conv(hidden_states)
+                vertical_outputs = self.vertical_batch_norm(vertical_outputs)
+            else:
+                vertical_outputs = 0
+
+            if self.horizontal_conv is not None:
+                horizontal_outputs = self.horizontal_conv(hidden_states)
+                horizontal_outputs = self.horizontal_batch_norm(horizontal_outputs)
+            else:
+                horizontal_outputs = 0
+
+            if self.rbr_identity is None:
+                id_out = 0
+            else:
+                id_out = self.rbr_identity(hidden_states)
+
+            return self.nonlinearity(main_outputs + vertical_outputs + horizontal_outputs + id_out)
+        else:
+            if not hasattr(self, "fused_conv"):
+                self.prepare_for_eval()
+            return self.nonlinearity(self.fused_conv(hidden_states))
+
+    def _identity_to_conv(self, identity):
+        if identity is None:
+            return 0, 0
+        if not hasattr(self, "id_tensor"):
+            input_dim = self.in_channels
+            kernel_value = np.zeros((self.in_channels, input_dim, 1, 1), dtype=np.float32)
+            for i in range(self.in_channels):
+                kernel_value[i, i % input_dim, 0, 0] = 1
+            id_tensor = torch.from_numpy(kernel_value).to(identity.weight.device)
+            self.id_tensor = self._pad_to_mxn_tensor(id_tensor)
+        kernel = self.id_tensor
+        running_mean = identity.running_mean
+        running_var = identity.running_var
+        gamma = identity.weight
+        beta = identity.bias
+        eps = identity.eps
+        std = (running_var + eps).sqrt()
+        t = (gamma / std).reshape(-1, 1, 1, 1)
+        return kernel * t, beta - running_mean * gamma / std
+
+    def _fuse_batch_norm_tensor(self, conv, batch_norm):
```

Comment:
A bunch of this logic is copied from above - you can abstract it out e.g. 

```suggestion
    def _fuse_batch_norm(self, kernel, batch_norm):
        running_mean = batch_norm.running_mean
        running_var = batch_norm.running_var
        gamma = batch_norm.weight
        beta = batch_norm.bias
        eps = batch_norm.eps
        std = (running_var + eps).sqrt()
        t = (gamma / std).reshape(-1, 1, 1, 1)
        return kernel * t, beta - running_mean * gamma / std
```